components/eam/src/physics/cam/physpkg.F90

```diff
@@ -1340,6 +1343,24 @@ subroutine phys_run2(phys_state, ztodt, phys_tend, pbuf2d,  cam_out, &
     call gmean_mass ('after tphysac FV:WET)', phys_state)
 #endif
 
+    !
+    ! Check for carbon conservation
+    !
+    do c = begchunk, endchunk
+       call get_total_carbon(phys_state(c), 'wet')
+    end do
+    call print_global_carbon_diags(phys_state, ztodt, nstep)
+    do c = begchunk, endchunk
+       ncol = get_ncols_p(c)
+       phys_state(c)%tc_prev(:ncol) = phys_state(c)%tc_curr(:ncol)
+       if (is_first_step()) then
+          phys_state(c)%tc_init(:ncol) = phys_state(c)%tc_curr(:ncol)
```

Comment:
I had not previously run any restart tests.  I performed one a few moments ago and it caused the model to crash immediately.  I will investigate what is happening there and address it.

Reply:
I think a run with co2_transport=false would also crash, since nans are used in init of variables. just a comment to run tests with co2_transport set to true and false.